v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/MigrateTableTransform.java

```diff
@@ -51,25 +52,26 @@ public class MigrateTableTransform extends PTransform<PBegin, PCollection<Void>>
   private ISchemaMapper schemaMapper;
   private ReaderImpl reader;
   private String shardId;
-  private String shardIdColumn;
   private SQLDialect sqlDialect;
 
+  private Map<String, String> srcTableToShardIdColumnMap;
+
   public MigrateTableTransform(
       SourceDbToSpannerOptions options,
       SpannerConfig spannerConfig,
       Ddl ddl,
       ISchemaMapper schemaMapper,
       ReaderImpl reader,
       String shardId,
-      String shardIdColumn) {
+      Map<String, String> srcTableToShardIdColumnMap) {
     this.options = options;
     this.spannerConfig = spannerConfig;
     this.ddl = ddl;
     this.schemaMapper = schemaMapper;
     this.reader = reader;
     this.shardId = StringUtils.isEmpty(shardId) ? "" : shardId;
-    this.shardIdColumn = shardIdColumn;
     this.sqlDialect = SQLDialect.valueOf(options.getSourceDbDialect());
+    this.srcTableToShardIdColumnMap = srcTableToShardIdColumnMap;
```

Comment:
`migrateTableTransform` is a table level.
Do we need to have the map here? Can we just use `ISchemaMapper schemaMapper` which is already available here to map it to the `shardIdColumn` directly and just pass it to the DLQ create?

Reply:
So the input comes across the tables in the current level. So this transform runs for a list of tables and the DLQ needs to identify based on which table the row is coming for.